cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
+	end = regexp.MustCompile(`(?i)\s+(-?(?:filetype|package|pkg|path|file)):(\S+)\s*$`)
+)
+
+func parseFilters(query *url.Values, qstr string, atstart bool) string {
+	q := *query
+
+	var re *regexp.Regexp
+	if atstart {
+		re = start
+	} else {
+		re = end
+	}
+
+	matches := re.FindStringSubmatchIndex(qstr)
+	for matches != nil {
+		// matches is [token_start, token_end, type_start, type_end, value_start, value_end]
+		filter := strings.ToLower(qstr[matches[2]:matches[3]])
+		value := qstr[matches[4]:matches[5]]
+
+		if strings.HasPrefix(filter, "-") {
```

Comment:
`file` is a synonym for `path`, so the filter is called `path` and `npath`. This case is not handled by the code as-is.